server/src/instant/grouped_queue.clj

```diff
@@ -5,7 +5,38 @@
    (java.util.concurrent.atomic AtomicInteger)
    (clojure.lang PersistentQueue)))
 
-(def empty-q PersistentQueue/EMPTY)
+;; ----------- 
+;; inflight-queue
+
+(def persisted-q-empty PersistentQueue/EMPTY)
+
+(def inflight-queue-empty
```

Comment:
After we get this right, can using deftype, so peek, pop, count, et al 'just work' (Clojure ftw)